4-custom-evaluations.ipynb

```diff
@@ -0,0 +1,744 @@
+      "\u001b[0m\u001b[1;3;34mRetrieving with query id 1f959cf9-5896-4210-9c12-14cb6e550b45: How do I get started?\n",
+      "\u001b[0mAnswer: To get started, you need to have a Datadog API key for LLM Observability. Additionally, if you are using the example script with OpenAI, you will need to set up an OpenAI API key stored in your environment as `OPENAI_API_KEY` and have the OpenAI Python library installed. You can refer to the OpenAI documentation for instructions on creating an API key and setting up Python.\n",
+      "Context: ['Prerequisites\\n\\n- LLM Observability requires a Datadog API key (see [the instructions for creating an API key][7]).\\n- The example script below uses OpenAI, but you can modify it to use a different provider. To run the script as written, you need:\\n    - An OpenAI API key stored in your environment as `OPENAI_API_KEY`. To create one, see [Account Setup][4] and [Set up your API key][6] in the OpenAI documentation.\\n    - The OpenAI Python library installed. See [Setting up Python][5] in the OpenAI documentation for instructions.']\n"
+     ]
+    }
+   ],
+   "source": [
+    "answer, _ = ask_docs(STARTER_QUESTION, retriever=recursive_retriever)\n",
+    "\n",
+    "print(\"Answer: {}\".format(answer))\n",
+    "print(\"Context: {}\".format([reference.text for reference in answer.source_nodes]))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Which response is better? Can we measure how much better a certain retrieval strategy is? Let's use RAGAS and Datadog to answer that question."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### RAGAS Setup\n",
+    "\n",
+    "RAGAS is an open source evaluation library for RAG based workflows. It's powered by LLM-assisted evaluations that the performance of your retrievals, augmented generation, and RAG workflow end-to-end."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install ragas --quiet"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First, we'll define a list of questions we'll ask our RAG app. Some RAGAS evaluations also require ground truth answers in relation to a target question, so we'll have to define those as well."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "eval_questions = [\n",
+    "    \"How do I get started?\",\n",
+    "    \"I have a complex chatbot, what root span should I use to represent this bot?\",\n",
+    "    \"I have a summarization llm service with some simple pre-and-post processing steps, what root span should I use to represent this bot?\",\n",
+    "    \"I don't want to manually instrument my app. Can I still use LLM Observability?\",\n",
+    "    \"What's the ml app tag?\",\n",
```

Comment:
```suggestion
    "    \"What's the ml-app tag?\",\n",
```

Reply:
I can't tell which of these format styles are intentional, so feel free to ignore any suggestions in these!